web_ui/frontend/components/layout/Sidebar/UserMenu.tsx

```diff
@@ -0,0 +1,109 @@
+import {RandomUserIcon} from "@/components/layout/Sidebar/RandomUserIcon";
+import {getUser} from "@/helpers/login";
+
+const UserMenu = () => {
+
+    const userMenuRef = React.useRef(null);
+
+    const {data: user, isLoading, error: fetchError, mutate} = useSWR(
+        "getUser",
+        getUser,
+        {
+            refreshInterval: 1000 * 60,
+            fallbackData: {authenticated: false}
+        }
+    )
+
+    const router = useRouter()
+
+    const [menuOpen, setMenuOpen] = useState(false)
+    const [error, setError] = useState<string | undefined>(undefined)
+
+    const handleLogout = async (e: React.MouseEvent<HTMLElement>) => {
+        try {
+            let response = await fetch("/api/v1.0/auth/logout", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                }
+            })
+
+            if(response.ok) {
+                await mutate(getUser)
+            } else {
+                try {
+                    let data = await response.json()
+                    if (data?.error) {
+                        setError(response.status + ": " + data['error'])
+                    } else {
+                        setError("Server error with status code " + response.status)
+                    }
+                } catch {
+                    setError("Server error with status code " + response.status)
+                }
+            }
+        } catch {
+            setError("Could not connect to server")
+        }
+    }
+
+    if(!user.authenticated){
+        return (
+            <IconButton
+                id={"user-menu-button"}
+                ref={userMenuRef}
+                sx={{
+                    bgcolor: "#767adb4a"
+                }}
+                onClick={() => router.push("/login" + "?returnURL=" + window.location.pathname)}
+            >
+                <Login/>
+            </IconButton>
+        )
+    }
+
+    return (
+        <>
+            <IconButton
+                id={"user-menu-button"}
+                ref={userMenuRef}
+                sx={{
+                    bgcolor: "#4dba5a3b"
+                }}
+                onClick={() => setMenuOpen(!menuOpen)}
+            >
+                <RandomUserIcon/>
+            </IconButton>
+            <Menu
+                id={"user-menu"}
+                aria-labelledby={"user-menu-button"}
+                sx={{ml:4}}
+                anchorEl={userMenuRef.current}
+                open={menuOpen}
+                onClose={() => setMenuOpen(false)}
+                anchorOrigin={{
+                    vertical: 'center',
+                    horizontal: 'right',
+                }}
+                transformOrigin={{
+                    vertical: 'center',
+                    horizontal: 'left',
+                }}
+            >
+                { user.role === "admin" ? <MenuItem disabled={true}>Admin</MenuItem> : null }
```

Comment:
Can we do a hover-over tooltip explaining what this means? something like "You have administrator privileges" or "You are an admin of this service" or changing the name to "Admin User"